rootless: add `Requires=dbus.socket`

AkihiroSuda opened the pull request:
- relates to https://github.com/moby/moby/issues/42793

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Added a dependency for dbus.

On a cgroup v2 host with systemd, dbus is needed to avoid the following error:
```
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed
: unable to start container process: unable to apply cgroup configuration: unable to start unit "docker-170a4183e351e69835b82cc3134b97c8cbb0e6d3a6
16d5a0fb0ea473075062ad.scope" (properties [{Name:Description Value:"libcontainer container 170a4183e351e69835b82cc3134b97c8cbb0e6d3a616d5a0fb0ea47
3075062ad"} {Name:Slice Value:"user.slice"} {Name:Delegate Value:true} {Name:PIDs Value:@au [2872]} {Name:MemoryAccounting Value:true} {Name:CPUAc
counting Value:true} {Name:IOAccounting Value:true} {Name:TasksAccounting Value:true} {Name:DefaultDependencies Value:false}]): Interactive authen
tication required.: unknown.
```


**- How I did it**

See the code

**- How to verify it**

```console
$ systemctl --user stop dbus.socket
$ systemctl --user is-active dbus.socket
inactive
$ systemctl --user start docker
$ systemctl --user is-active dbus.socket
active
```

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section:
-->
```markdown changelog
rootless: add `Requires=dbus.socket`
```

**- A picture of a cute animal (not mandatory but encouraged)**
🐧

thaJeztah commented on contrib/dockerd-rootless-setuptool.sh:
```diff
@@ -264,6 +264,19 @@ init() {
+		# dbus should be automatically enabled on installing dbus-user-session package
+		# (depended by docker-ce-rootless-extras), but dbus does not start automatically
+		# until the next login.
+		ERROR "dbus user session is not enabled. Run the following command and run this tool again."
```
Curious; is this something that the systemd unit that we run could declare as dependency, and automatically take care of?

AkihiroSuda replied:
Yes, updated

thaJeztah commented on contrib/dockerd-rootless-setuptool.sh:
```diff
@@ -264,6 +264,19 @@ init() {
+	if [ -n "$SYSTEMD" ] && ! systemctl --user -q is-active dbus; then
+		# dbus should be automatically enabled on installing dbus-user-session package
+		# (depended by docker-ce-rootless-extras), but dbus does not start automatically
+		# until the next login.
```
Oh! Because of this? That's odd; how's that normally supposed to be done (require logout and login again?)

AkihiroSuda replied:
Now `systemctl --user start docker` starts dbus

AkihiroSuda committed:
rootless: add `Requires=dbus.socket`

On a cgroup v2 host with systemd, dbus is needed to avoid the following error:
```
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed
: unable to start container process: unable to apply cgroup configuration: unable to start unit "docker-170a4183e351e69835b82cc3134b97c8cbb0e6d3a6
16d5a0fb0ea473075062ad.scope" (properties [{Name:Description Value:"libcontainer container 170a4183e351e69835b82cc3134b97c8cbb0e6d3a616d5a0fb0ea47
3075062ad"} {Name:Slice Value:"user.slice"} {Name:Delegate Value:true} {Name:PIDs Value:@au [2872]} {Name:MemoryAccounting Value:true} {Name:CPUAc
counting Value:true} {Name:IOAccounting Value:true} {Name:TasksAccounting Value:true} {Name:DefaultDependencies Value:false}]): Interactive authen
tication required.: unknown.
```

Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

thaJeztah reviewed (approved):
LGTM

curious if there's should also be some "After=" and "Wants=" similar to the non-rootless variant, but I guess we can add some of those later if they would be good to add
